placementApp/views.py

```diff
@@ -1,14 +1,30 @@
 import xlwt
 import datetime
 from django.shortcuts import HttpResponse
-from .models import Student, Position, Company
+from .models import Student, Position, Company, Application
 from .serializers import *
 from django.contrib.auth import get_user_model
 from rest_framework import viewsets, permissions, status, mixins, generics
 from django.contrib.auth.hashers import make_password
 from rest_framework.response import Response
 
 
+class ApplicationViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    permission_classes = (permissions.AllowAny,)
+    queryset = Application.objects.all()
+    serializer_class = ApplicationSerializer
+
+    def perform_create(self, serializer):
+        serializer.save(student=Student.objects.get(pk=2))
```

Comment:
No need of hardcoding pk. use self.request.user.pk. This will work when you are signed in as a student. Use swagger ui testing. "localhost:8000/docs"  url. This will help you figure out what is not correct with Application serializer.

Reply:
Have made the changes